### PR TITLE
Gallery view

### PR DESCRIPTION
**Pull request by jespino**

### CODE REVIEW

File: webapp/src/components/sidebar/sidebarBoardItem.tsx [hunk 1]
```diff
@@ -35,6 +36,7 @@ const SidebarBoardItem = React.memo((props: Props) => {
         switch (viewType) {
         case 'board': return <BoardIcon/>
         case 'table': return <TableIcon/>
+        case 'gallery': return <ImageIcon/>
```
Comment (chenilim) [inline webapp/src/components/sidebar/sidebarBoardItem.tsx:39]: We can fix this later, but let's add a new `GalleryIcon` class, copied from ImageIcon, then assign a new issue to Assad to create a new icon.

File: webapp/src/components/gallery/galleryCard.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    const {card} = props
+    const [cardTree, setCardTree] = useState<CardTree>()
+    useCardListener(
+        card.id,
+        async (blocks) => {
+            const newCardTree = cardTree ? MutableCardTree.incrementalUpdate(cardTree, blocks) : await MutableCardTree.sync(card.id)
+            setCardTree(newCardTree)
+        },
+        async () => {
+            const newCardTree = await MutableCardTree.sync(card.id)
```
Comment (chenilim) [inline webapp/src/components/gallery/galleryCard.tsx:32]: This is fine for now, but some thoughts on how we could reduce the number of API calls in the future:
* Let the Gallery own the state
* Initially we could modify / extend BoardTree to call getSubTree with 3 levels
* I envision adding a `getTree` API that returns all nodes by rootID
* And updating the websocket to notify changes to trees by rootID as well
* Then state can be managed by a `fullBoardTree` view model
* In the future-future, we can look at adding filters to the query (e.g. to limit types of nodes), or build something closer to GraphQL.

Just initial thoughts.

**Review by chenilim — APPROVED** [review]
I think folks will be very excited to see this!

File: webapp/src/components/sidebar/sidebarBoardItem.tsx (hunk 1 above)
Comment (jespino) [reply webapp/src/components/sidebar/sidebarBoardItem.tsx]: Done

**Pull request merged**